src/test/groovy/org/sonarqube/gradle/FunctionalTests.groovy

```diff
@@ -233,6 +233,64 @@ class FunctionalTests extends Specification {
         props."sonar.java.target" == '8'
     }
 
+    def "log execution context"() {
+        given:
+        settingsFile << "rootProject.name = 'java-task-output-logs'"
+        buildFile << """
+        plugins {
+            id 'java'
+            id 'org.sonarqube'
+        }
+        """
+
+        when:
+        def result = GradleRunner.create()
+          .withGradleVersion(gradleVersion)
+          .withProjectDir(projectDir.toFile())
+          .withEnvironment(Map.of("GRADLE_OPTS", "-Dfoo=bar"))
+          .forwardOutput()
+          .withArguments('sonarqube', '--info', '-Dsonar.scanner.internal.dumpToFile=' + outFile.toAbsolutePath())
+          .withPluginClasspath()
+          .build()
+
+        then:
+        result.task(":sonarqube").outcome == SUCCESS
+        nonNull(
+          assertThat(result.output)
+            .containsPattern('org.sonarqube Gradle plugin \\d+\\.\\d+')
+            .containsPattern('Java \\d+')
+            .contains('(64-bit)')
+            .contains('GRADLE_OPTS=-Dfoo=bar')
+        )
+    }
+
+    def "does not log execution context when sonar.skip is true"() {
```

Comment:
I think we should still print the message even if the project/subproject is skipped